onnxruntime/core/mlas/lib/quantize.cpp

```diff
@@ -805,6 +947,58 @@ MlasQuantizeLinear<uint16_t>(
     uint16_t ZeroPoint
     );
 
+template <bool Signed>
+void
+MLASCALL
+MlasQuantizeLinearInt4(
+    const float* Input,
+    uint8_t* Output,
+    size_t N,
+    float Scale,
+    int8_t ZeroPoint
+    )
+{
+    constexpr int32_t MinimumValue = Int4Traits<Signed>::Min;
+    constexpr int32_t MaximumValue = Int4Traits<Signed>::Max;
+    using UnpackedType = typename Int4Traits<Signed>::UnpackedType;
+
+    for (size_t n = 0; n < N; n++) {
+        float FloatValue = std::nearbyintf(Input[n] / Scale) + static_cast<float>(ZeroPoint);
```

Comment:
will std::nearbyintf round to nearest even here? assuming we want that mode as it's specified for ONNX QuantizeLinear
https://github.com/onnx/onnx/blob/093a8d335a66ea136eb1f16b3a1ce6237ee353ab/docs/Operators.md?plain=1#L20288